### ISSUE TITLE
Make Tangram mobile friendly

### ISSUE THREAD

**Issue by tchx84** [sonnyp/Tangram#87]
Great app!
 
Would be great if Tangram could maximize screen space for the web apps on small devices by doing one of the following: 

a) Have an option to hide the left applications list, so each web app can be fit on small displays, or even better

b) Use a HdyLeaflet https://blogs.gnome.org/alexm/2019/09/15/gnome-and-gestures-part-2-hdyleaflet/#:~:text=A%20folded%20HdyLeaflet%20%2C%20just%20like,and%20which%20is%20a%20screenshot%3F

**Comment by tchx84** [comment]
Oh, I noticed now that you can actually move the tabs position :) heh, turns out I could see that because the windows can't shrink below a certain minimum size, maybe that size can be removed? So it properly fits in small displays?

**Comment by sonnyp** [comment]
I don't really have a use (yet?) for it myself but I'd welcome a PR.

I'm ok with trading tabs (including the horizontal option) for a HdyLeaflet if it means better small window UX.

Something like https://blogs.gnome.org/alexm/files/2019/09/swipe-demo6.gif where the sidebar is visible on wider screen.

> Great app!

Back at you, :heart: Flatseal!

**Comment by sonnyp** [comment]
I dig a bit and moving away from [GtkNotebook](https://developer.gnome.org/gtk3/stable/GtkNotebook.html) means we would loose

##### re-ordering tabs

We need an alternative

##### tab bar positions

I'm ok with loosing that.

##### moving tabs between windows

Not currently in used but insteresting for #21. Let's see when we have virtual applications but I'd be ok with a "send tab to" option. It also works better on mobile.

**TLDR**; I'm cool with replacing GTKNotebook with a more friendly mobile solution but it needs to support re-ordering somehow.

**Comment by tchx84** [comment]
> I don't really have

You can, literally, just download the x86_64 image for Librem 5 and boot it in Boxes, see https://developer.puri.sm/Librem5/Development_Environment/Boards/qemu.html (ignore the instructions, just download the image from there)

> Also cool if the PR just removes the minimum window size.

After closer inspection, the issue is that there's too many widgets in the header bar. So, even if it switches to a HdyLeaflet it simply won't shrink.

I had a similar issue with Flatseal, and what I ended up doing is to move some of the options to an action bar at the bottom, but only when needed. See https://github.com/tchx84/Flatseal/blob/master/screenshots/en/3.png

> Back at you, heart Flatseal!

Hah, thanks!

**Comment by sonnyp** [comment]
> You can, literally, just download the x86_64 image for Librem 5 and boot it in Boxes, see https://developer.puri.sm/Librem5/Development_Environment/Boards/qemu.html (ignore the instructions, just download the image from there)

I didn't mean to say I can't use/test such feature but rather that I don't have a use case for it myself and because of that I probably won't implement this myself in the near future.

Happy to help if anyone is willing to work on this though.

**Comment by tchx84** [comment]
> I didn't mean to say I can't use/test such feature but rather that I don't have a use case for it myself and because of that I probably won't implement this myself in the near future.

My bad!
 
> Happy to help if anyone is willing to work on this though.

Fair enough :)

**Comment by BrainBlasted** [comment]
Hey there - you may be interested in the work going on at https://gitlab.gnome.org/Teams/Design/os-mockups/-/issues/49 :) There's ongoing work to make a new adaptive tab widget and tab view for mobile form factors.

**Comment by sonnyp** [comment]
@BrainBlasted fantastic! thank you.

**Comment by BrainBlasted** [comment]
In the meantime, you could take a look at what Ephy currently uses. See my MRs here:

* https://gitlab.gnome.org/GNOME/epiphany/-/merge_requests/275
* https://gitlab.gnome.org/GNOME/epiphany/-/merge_requests/282
* https://gitlab.gnome.org/GNOME/epiphany/-/merge_requests/283
